src/main/java/Bro.java

```diff
@@ -99,8 +99,84 @@ public void run() {
+     * @param ui      An instance of Ui used for generating user interface messages.
+     * @param tasks   The TaskList containing the current list of tasks.
+     * @param storage An instance of Storage used for saving tasks to a file.
+     * @return        A string response generated based on the input command. This could be a list of tasks,
+     *                a confirmation message, or an error message depending on the command.
+     * @throws BroException            If there is a specific error related to task operations.
+     * @throws NumberFormatException   If the input for task indexing is not a valid number.
+     * @throws IndexOutOfBoundsException If the task index specified is out of range.
+     */
+    public String getResponse(String word, Ui ui, TaskList tasks, Storage storage) {
+        if (word.equalsIgnoreCase("list")) {
+            return ui.printList(tasks);
+        } else {
+            String action = word.split(" ", 2)[0];
+            String info;
+            if (word.split(" ", 2).length == 1) {
+                info = "";
+            } else {
+                info = word.split(" ", 2)[1];
+            }
+            try {
+                switch (action.toLowerCase()) {
+                    case "mark":
+                        String mark = tasks.markTask(Integer.parseInt(info));
+                        storage.saveToFile();
+                        return mark;
+                        //break;
+                    case "unmark":
+                        String unmark = tasks.unmarkTask(Integer.parseInt(info));
+                        storage.saveToFile();
+                        return unmark;
+                        //break;
+                    case "delete":
+                        String delete = tasks.deleteTask(Integer.parseInt(info));
+                        storage.saveToFile();
+                        return delete;
+                        //break;
+                    case "find":
+                        return tasks.findTasks(info);
+                        //break;
+                    case "todo":
+                        String todo = tasks.addTodo(info);
+                        storage.saveToFile();
+                        return todo;
+                        //break;
+                    case "deadline":
+                        String deadline = tasks.addDeadline(info);
+                        storage.saveToFile();
+                        return deadline;
+                        //break;
+                    case "event":
+                        String event = tasks.addEvent(info);
+                        storage.saveToFile();
+                        return event;
+                        //break;
+                    default:
+                        return ui.printDefault();
+                }
+            } catch (BroException be) {
+                return be.getMessage();
+            } catch (NumberFormatException nfe) {
```

Comment:
I like how you've caught specific exceptions like BroException, NumberFormatException, and IndexOutOfBoundsException.